samples/python_interop/cirq_samples/teleportation_basics.ipynb

```diff
@@ -0,0 +1,188 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ade75f5f",
+   "metadata": {},
+   "source": [
+    "## 1. Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a8201f1b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cirq\n",
+    "from typing import Sequence"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "deeaace0",
+   "metadata": {},
+   "source": [
+    "## 2. Helper Functions\n",
+    "We define small helpers for Bell pair preparation and message encoding."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4557aa98",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def PrepareBellPair(q1: cirq.Qid, q2: cirq.Qid) -> Sequence[cirq.Operation]:\n",
+    "    return [\n",
+    "        cirq.H.on(q1),\n",
+    "        cirq.CNOT.on(q1, q2)\n",
+    "    ]\n",
+    "\n",
+    "def PrepareMessage(alice: cirq.Qid, message: cirq.Qid) -> Sequence[cirq.Operation]:\n",
+    "    return [\n",
+    "        cirq.CNOT.on(message, alice),\n",
+    "        cirq.H.on(message)\n",
+    "    ]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b9e379d2",
+   "metadata": {},
+   "source": [
+    "## 3. Building the Teleportation Operation List\n",
+    "We build the protocol step-by-step."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3be71578",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def simple_teleportation() -> Sequence[cirq.Operation]:\n",
+    "    alice = cirq.NamedQubit('alice')\n",
+    "    bob = cirq.NamedQubit('bob')\n",
+    "    program: Sequence[cirq.Operation] = []\n",
+    "    # 1. Create entangled pair shared by Alice and Bob\n",
+    "    program.extend(PrepareBellPair(alice, bob))\n",
+    "    # 2. Prepare a message qubit (state to teleport)\n",
+    "    msg = cirq.NamedQubit('msg')\n",
+    "    program.append(cirq.rx(0.7)(msg))  # Example arbitrary state\n",
+    "    # 3. Entangle message with Alice's half\n",
+    "    program.extend(PrepareMessage(alice, msg))\n",
+    "    # 4. Measure Alice + msg to obtain classical bits\n",
+    "    program.append(cirq.measure(msg, key='b1'))\n",
+    "    program.append(cirq.measure(alice, key='b2'))\n",
+    "    # 5. Classical feed-forward corrections on Bob\n",
+    "    program.append(cirq.Z.on(bob).with_classical_controls('b1'))\n",
+    "    program.append(cirq.X.on(bob).with_classical_controls('b2'))\n",
+    "    # 6. Measure Bob's qubit directly to sample teleported state\n",
+    "    program.append(cirq.measure(bob, key='result'))\n",
+    "    return program"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2882e3d0",
+   "metadata": {},
+   "source": [
+    "## 4. Circuit Construction and Display"
```

Comment:
This is the first time moment marks appear on the diagram. Suggest commenting what they are.

Reply:
I added a more detailed description of moments in cirq and how they appear. Let me know if that description makes sense.